ci/bootstrap_create_accounts.py

```diff
@@ -0,0 +1,81 @@
+import os
+import base64
+import json
+import kubernetes_asyncio as kube
+from hailtop import aiogoogle
+from hailtop.utils import async_to_blocking
+from gear import Database
+
+from auth.driver.driver import create_user
+
+SCOPE = os.environ['HAIL_SCOPE']
+PROJECT = os.environ['HAIL_PROJECT']
+DEFAULT_NAMESPACE = os.environ['HAIL_DEFAULT_NAMESPACE']
+
+async def insert_user_if_not_exists(app, username, email, is_developer, is_service_account):
+    db = app['db']
+    k8s_client = app['k8s_client']
+
+    row = await db.execute_and_fetchone('SELECT id, state FROM users where username = %s;', (username,))
+    if row:
+        if row['state'] == 'active':
+            return None
+        return row['id']
+
+    gsa_key_secret_name = f'{username}-gsa-key'
+
+    secret = await k8s_client.read_namespaced_secret(gsa_key_secret_name, DEFAULT_NAMESPACE)
+    key_json = base64.b64decode(secret.data['key.json']).decode()
+    key = json.loads(key_json)
+    gsa_email = key['client_email']
+
+    if is_developer and SCOPE != 'deploy':
+        namespace_name = DEFAULT_NAMESPACE
+    else:
+        namespace_name = None
+
+    return await db.execute_insertone(
+        '''
+INSERT INTO users (state, username, email, is_developer, is_service_account, gsa_email, gsa_key_secret_name, namespace_name)
+VALUES (%s, %s, %s, %s, %s, %s, %s, %s);
+''',
+        ('creating', username, email, is_developer, is_service_account, gsa_email, gsa_key_secret_name, namespace_name))
```

Comment:
This function races between attempts of this job. I think we ought to do an `insert ... on duplicate key ignore` followed by a select to get the user's id. `create_user` looks easy to make idempotent. We can race to create a session, it doesn't matter which one ends up in the secret as long as it's a valid one. All the k8s secret creates need to be idempotent create.

Reply:
I handled this slightly differently.  I already check the user exists above, but you're right there is still a race condition (two racing inserts) so I wrapped this in a transaction.